torch/distributed/distributed_c10d.py

```diff
@@ -718,6 +718,8 @@ def irecv(tensor,
     if group == GroupMember.WORLD:
         _check_default_pg()
         return _default_pg.recv([tensor], src, tag)
+    elif src is None:
+        return group.recv_anysource([tensor], tag)
```

Comment:
Similar to `recv`, shouldn't this work irrespective of whether the group is WORLD or not?

Reply:
Fixed